File: src/main/java/dominio/indicadores/ExpresionBuilder.java
package dominio.indicadores;

public class ExpresionBuilder {
	
	private Expresion expresion;
	
	public Expresion build(){
		//Se podrían agregar validaciones
		return expresion;
	}
	
	public void agregarExpresion(Expresion expresion){
		this.expresion = expresion;
	}
	
	public void agregarOperacion(Expresion operandoDer, String operadorBinario){
		expresion = new ExpresionOperacion(expresion, operandoDer, operadorBinario);
	}
	
}

File: src/main/java/dominio/indicadores/ExpresionBuilder.kt
package dominio.indicadores

class ExpresionBuilder {

    private var expresion: Expresion? = null

    fun build(): Expresion = expresion!!

    fun agregarExpresion(expresion: Expresion) {
        this.expresion = expresion
    }

    fun agregarOperacion(operandoDer: Expresion, operadorBinario: String) {
        expresion = ExpresionOperacion(expresion!!, operandoDer, operadorBinario)
    }

}
